packit_service/models.py

```diff
@@ -2495,9 +2497,28 @@ def get_first_successful_by_sidetag(
                     BodhiUpdateTargetModel.status == "success",
                     BodhiUpdateTargetModel.sidetag == sidetag,
                 )
+                .order_by(BodhiUpdateTargetModel.update_creation_time.desc())
                 .first()
             )
 
+    @classmethod
+    def get_all_successful_or_in_progress_by_nvrs(
+        cls,
+        koji_nvrs: str,
+    ) -> set["BodhiUpdateTargetModel"]:
+        regexp = "|".join(re.escape(nvr) for nvr in set(koji_nvrs.split()))
+        with sa_session_transaction() as session:
+            return set(
+                session.query(BodhiUpdateTargetModel)
+                .filter(
+                    BodhiUpdateTargetModel.status.in_(("queued", "retry", "success")),
+                    BodhiUpdateTargetModel.koji_nvrs.regexp_match(regexp).compile(
```

Comment:
Does this actually precompile?

Reply:
Not sure, I followed  an example from the [documentation](https://docs.sqlalchemy.org/en/20/core/operators.html#string-matching).